cms/djangoapps/contentstore/views/certificates.py

```diff
@@ -276,7 +269,11 @@ def certificates_detail_handler(request, course_key_string, certificate_id):
     store = modulestore()
     with store.bulk_operations(course_key):
         course = get_course_and_check_access(course_key, request.user)
-        certificates_list = course.certificates['certificates']
+        certificates_list = []
+        if course.certificates.get('certificates') is not None:
+            certificates_list = course.certificates['certificates']
+        else:
+            course.certificates['certificates'] = []
 
```

Comment:
Looking good so far, although Jenkins appears to have failed on some
quality violations -- the ones pertaining to unused imports in the
certificates views are expected, but the others we should address.

On Tue, Apr 14, 2015 at 10:25 AM, Asad Iqbal <notifications@github.com>
wrote:

> In cms/djangoapps/contentstore/views/certificates.py
> <https://github.com/edx/edx-platform/pull/7681#discussion_r28330979>:
>
> > @@ -276,7 +269,11 @@ def certificates_detail_handler(request, course_key_string, certificate_id):
> >      store = modulestore()
> >      with store.bulk_operations(course_key):
> >          course = get_course_and_check_access(course_key, request.user)
> > -        certificates_list = course.certificates['certificates']
> > +        certificates_list = []
> > +        if course.certificates.get('certificates') is not None:
> > +            certificates_list = course.certificates['certificates']
> > +        else:
> > +            course.certificates['certificates'] = []
> >
>
> just for safe side i have added this check. i know in edit certificate
> scenario , we have already a poplutted dict under course.certificates.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/7681/files#r28330979>.
>



-- 
Matthew Drayer
Solutions Architect
edX | MIT
mattdrayer@edx.org